### PR TITLE
docs: add react native SDK manual

### PR DESCRIPTION
**Pull request by zhu-xiaowei**
----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

## Summary

1. add react native SDK manual

## Implementation highlights

(describe how the merge request does for feature changes, share the RFC link if it has)

## Test checklist

- [ ] add new test cases
- [ ] all code changes are covered by unit tests
- [ ] end-to-end tests
  - [ ] deploy web console with CloudFront + S3 + API gateway
  - [ ] deploy web console within VPC
  - [ ] deploy ingestion server
    - [ ] with MSK sink
    - [ ] with KDS sink
    - [ ] with S3 sink
  - [ ] deploy data processing
  - [ ] deploy data modeling
    - [ ] new Redshift Serverless
    - [ ] provisioned Redshift
    - [ ] Athena
  - [ ] deploy with reporting
  - [ ] streaming ingestion
    - [ ] with Redshift Serverless
    - [ ] with provisioned Redshift

## Is it a breaking change

- [ ] add parameters without default value in stack
- [ ] introduce new service permission in stack
- [ ] introduce new top level stack module

## Miscellaneous

- [ ] introduce new symbol link source file(s) to be shared among infra code, web console frontend, and web console backend

### CODE REVIEW

File: docs/en/sdk-manual/flutter.md [hunk 1]
```diff
@@ -108,9 +108,13 @@ analytics.setUserAttributes({
 });
 ```
 
-Current login user's attributes will be cached in disk, so the next time app launch you don't need to set all user's
-attribute again, of course you can use the same API `analytics.setUserAttributes()` to update the current user's
-attribute when it changes.
+Current login user's attributes will be cached in disk, so the next time app open you don't need to set all user
```
Comment (luorobin-a2z) [inline docs/en/sdk-manual/flutter.md:111]: login -> logged-in

File: docs/en/sdk-manual/react-native.md [hunk 2]
```diff
@@ -0,0 +1,376 @@
...
+```typescript
+import { ClickstreamAnalytics } from '@aws/clickstream-react-native';
+
+ClickstreamAnalytics.setUserAttributes({
+  userName: "carl",
+  userAge: 22
+});
+```
+
+Current login user's attributes will be cached in disk, so the next time app open you don't need to set all user
```
Comment (luorobin-a2z) [inline docs/en/sdk-manual/react-native.md:132]: login -> logged-in

File: docs/en/sdk-manual/react-native.md [hunk 3]
```diff
@@ -0,0 +1,376 @@
...
+
+By default, SDK will automatically track the preset `_screen_view` event when Android Activity triggers `onResume` or
+iOS ViewController triggers `viewDidAppear`.
+
+You can also manually record screen view events whether automatic screen view tracking is enabled, add the following
+code to record a screen view event with two attributes.
+
+* `SCREEN_NAME` Required. Your screen's name.
+* `SCREEN_UNIQUE_ID` Optional. Set the id of your component. If you do not set, the SDK will set a default value based
+  on the hashcode of the current native Activity or native ViewController.
```
Comment (luorobin-a2z) [inline docs/en/sdk-manual/react-native.md:182]: so SCREEN ID is not required for manual tracking screen view? but should we allow user user to set it optionally if they want to

File: docs/en/sdk-manual/react-native.md [hunk 4]
```diff
@@ -0,0 +1,376 @@
...
+```typescript
+import { ClickstreamAnalytics } from '@aws/clickstream-react-native';
+
+ClickstreamAnalytics.updateConfigure({
+  appId: 'your appId',
+  endpoint: 'https://example.com/collect',
+  isLogEvents: true,
+  authCookie: 'your auth cookie',
+  isCompressEvents: true,
+  isTrackPageViewEvents: false,
```
Comment (luorobin-a2z) [inline docs/en/sdk-manual/react-native.md:259]: isTrackScreenViewEvent?

File: docs/zh/sdk-manual/react-native.md [hunk 5]
```diff
@@ -0,0 +1,365 @@
...
+import { ClickstreamAnalytics } from '@aws/clickstream-react-native';
+
+ClickstreamAnalytics.updateConfigure({
+  appId: 'your appId',
+  endpoint: 'https://example.com/collect',
+  isLogEvents: true,
+  authCookie: 'your auth cookie',
+  isCompressEvents: true,
+  isTrackPageViewEvents: false,
+  isTrackUserEngagementEvents: false,
```
Comment (luorobin-a2z) [inline docs/zh/sdk-manual/react-native.md:253]: -->isTrackScreenViewEvents?

**Review by luorobin-a2z — COMMENTED** [review]
with a few comments and wording suggestions

File: docs/zh/sdk-manual/react-native.md (hunk 5 above)
Comment (zhu-xiaowei) [reply docs/zh/sdk-manual/react-native.md:253]: fixed.

File: docs/en/sdk-manual/react-native.md (hunk 4 above)
Comment (zhu-xiaowei) [reply docs/en/sdk-manual/react-native.md:259]: fixed.

File: docs/en/sdk-manual/react-native.md (hunk 2 above)
Comment (zhu-xiaowei) [reply docs/en/sdk-manual/react-native.md:132]: fixed.

File: docs/en/sdk-manual/flutter.md (hunk 1 above)
Comment (zhu-xiaowei) [reply docs/en/sdk-manual/flutter.md:111]: fixed.

File: docs/en/sdk-manual/react-native.md (hunk 3 above)
Comment (zhu-xiaowei) [reply docs/en/sdk-manual/react-native.md:182]: The meaning of screen id here is the class path of the screen. The user can set it through add additional custom attributes if needed. For App screen view, I think these two common attributes are enough.

File: docs/en/sdk-manual/react-native.md [hunk 6]
```diff
@@ -0,0 +1,376 @@
...
+     appId: 'your appId',
+     endpoint: 'https://example.com/collect',
+     isLogEvents: true,
+   });
+   ```
+
+2. After configuring `isLogEvents:true`, when you record an event, you can see the event raw
+   json in AndroidStudio Logcat or Xcode debug console by filter `EventRecorder`.
+
+## Data format definition
```
Comment (zxkane) [inline docs/en/sdk-manual/react-native.md:298]: Is it the same as the web SDK or Android/iOS SDK? If they are same, let's add a link to other SDK section for maintaining duplicated contents.

**Review by zxkane — CHANGES_REQUESTED** [review]

File: docs/en/sdk-manual/react-native.md (hunk 6 above)
Comment (zhu-xiaowei) [reply docs/en/sdk-manual/react-native.md:298]: added link

File: docs/en/sdk-manual/react-native.md [hunk 7]
```diff
@@ -0,0 +1,376 @@
...
+
+```typescript
+import { ClickstreamAnalytics } from '@aws/clickstream-react-native';
+
+ClickstreamAnalytics.flushEvents();
+```
+
+#### Other configurations
+
+In addition to the required `appId` and `endpoint`, you can configure other information to get more customized usage
```
[the remark this answers is not in the document]
Comment (zhu-xiaowei) [reply docs/en/sdk-manual/react-native.md:206]: fixed.

**Review by zxkane — APPROVED** [review]

**Pull request merged**